src/magnify/apps/core/permissions.py

```diff
@@ -63,16 +65,24 @@ def has_object_permission(self, request, view, obj):
             return True
 
         user = request.user
+
+        if request.method == "DELETE":
+            return obj.user_accesses.filter(
+                role=UserRoleChoices.OWNER, user=user
+            ).exists()
```

Comment:
I'm not sure it is a good pattern for a permission to manage different permissions. It would be more explicit if a second permission was used at the viewset level according to the action.

Reply:
I'm not sure either actually! I would say it's nice to handle all permission stuff in this class and have one permission class per specific endpoint? But anyway for certain permissions like field-by-field update permissions, I have to handle it in the serializer...

Reply:
After playing a bit with it I think it makes it harder to split permission code between the viewset and the permission class... 
It may be a better pattern when we have simple permission checks like IsStaff, IsAuthenticated etc. that we can compose.
But in the case of RBAC like here it is probably easier to have one permission class per object.... and a lot of tests!!